src/common/slices/iconSlice.ts

```diff
@@ -0,0 +1,121 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { getServiceClient, URLS } from '../services';
+import { typeIconInfos } from './icons';
+
+export interface IconInfo {
+  icon?: string;
+  color: string;
+  url?: string;
+  isImage: boolean;
+}
```

Comment:
this type seems a little strange. Maybe consider the following to prevent having to check that icon/url are defined after checking isImage (as, if i'm reading this correctly, `isImage` tells you which should be defined)

```suggestion
export type IconInfo = {
    isImage: true;
    url: string;
    color: string;
} | {
    isImage: false;
    icon: string;
    color: string;
};
```

Reply:
When I try to do this, trypescript blows up because its expecting one type or another in different contexts.